smartcontractkit/external-adapters-js
# Upgrade Redis to V4

## Closes #23048

## Description
This PR upgrades Redis from V3 to V4, which had some moderate changes, mostly to the `createClient` interface. You can read about the V3 to V4 notes [here](https://github.com/redis/node-redis/blob/master/docs/v3-to-v4.md), and the new `createClient` config [here](https://github.com/redis/node-redis/blob/master/docs/client-configuration.md).

## Changes
- Update `createClient` options interface
- Add new type imports (hacky solution for now, but hopefully this can be fixed in the future)
- Simplify the new `reconnectStrategy` from `retryStrategy` since the new version only allows the number of retries as an input argument, without any other data
- `await client.connect()` to explicitly connect the client to the Redis server



## Steps to Test

The only tests I've run so far are confirming that the Redis server can be used without failing outright. No stress testing or confirming specific Redis configuration params.
1. `yarn && yarn setup`
2. `cd packages/sources/<adapter-to-test> && yarn start`
3. In separate terminal, run ` docker run -p 6379:6379 --name ea-redis -d redis redis-server`
4. Run a query for the adapter twice to confirm it caches the first time, and uses the cached result the second time

Ex. 
`yarn && yarn setup && cd packages/sources/coinbase && yarn start`, then in a separate window,
`curl -X POST http://localhost:8080 -H "Content-Type: application/json" -d '{"jobRunID": "1", "data": {"from": "BTC","to": "USD"}}'`
**First response:**
`{"jobRunID":"1","result":50792.59,"debug":{"staleness":0,"performance":0.192739083,"providerCost":1},"statusCode":200,"data":{"data":{"base":"BTC","currency":"USD","amount":"50792.59"},"result":50792.59}}`
**Second response (after running the curl command again):**
`{"result":50792.59,"maxAge":30000,"debug":{"cacheHit":true,"staleness":0.854,"performance":0.009729041,"providerCost":0},"statusCode":200,"data":{"data":{"base":"BTC","currency":"USD","amount":"50792.59"},"result":50792.59}}`

## Quality Assurance
